src/schemathesis/runner/__init__.py

```diff
@@ -1,116 +1,155 @@
-    for option in produces:
-        if are_content_types_equal(option, content_type):
-            return
-    raise AssertionError(
-        f"Received a response with '{content_type}' Content-Type, "
-        f"but it is not declared in the schema.\n\n"
-        f"Defined content types: {', '.join(produces)}"
-    )
+        initialized = events.Initialized(
+            results=results, schema=self.schema, checks=self.checks, hypothesis_settings=self.hypothesis_settings
+        )
+        yield initialized
 
+        yield from self._execute(results)
 
-def response_schema_conformance(response: requests.Response, result: TestResult) -> None:
-    if not response.headers["Content-Type"].startswith("application/json"):
-        return
-    # the keys should be strings
-    responses = {str(key): value for key, value in result.endpoint.definition.get("responses", {}).items()}
-    status_code = str(response.status_code)
-    if status_code in responses:
-        definition = responses[status_code]
-    elif "default" in responses:
-        definition = responses["default"]
-    else:
-        # No response defined for the received response status code
-        return
-    schema = definition.get("schema")
-    if not schema:
-        return
-    try:
-        jsonschema.validate(response.json(), schema)
-    except jsonschema.ValidationError as exc:
-        raise AssertionError(f"The received response does not conform to the defined schema!\n\nDetails: \n\n{exc}")
+        yield events.Finished(results=results, schema=self.schema, running_time=time.time() - initialized.start_time)
 
+    def _execute(self, results: TestResultSet) -> Generator[events.ExecutionEvent, None, None]:
+        raise NotImplementedError
 
-DEFAULT_CHECKS = (not_a_server_error,)
-OPTIONAL_CHECKS = (status_code_conformance, content_type_conformance, response_schema_conformance)
-ALL_CHECKS: Tuple[Callable[[Response, TestResult], None], ...] = DEFAULT_CHECKS + OPTIONAL_CHECKS
 
+class InProcessRunner(BaseRunner):
+    def _execute(self, results: TestResultSet) -> Generator[events.ExecutionEvent, None, None]:
+        with get_session(self.auth, self.headers) as session:
+            for endpoint, test in self.schema.get_all_tests(single_test, self.hypothesis_settings, self.seed):
+                for event in run_test(self.schema, endpoint, test, self.checks, session, results, self.request_timeout):
+                    yield event
+                    if isinstance(event, events.Interrupted):
+                        return
 
-@contextmanager
-def get_session(
-    auth: Optional[Auth] = None, headers: Optional[Dict[str, Any]] = None
-) -> Generator[requests.Session, None, None]:
-    with requests.Session() as session:
-        if auth is not None:
-            session.auth = auth
-        session.headers["User-agent"] = USER_AGENT
-        if headers is not None:
-            session.headers.update(**headers)
-        yield session
 
+class Worker(threading.Thread):
```

Comment:
Maybe, default `Thread` will be easier, just with passing everything needed to the execution:

```
Thread(target = run, args = (...))
```